tests/installation/module_registration/add_ha_regcode.pm

```diff
@@ -0,0 +1,23 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: After register Application module
+#          in "Extension and Module Selection" dialog
+#          input register code at register page
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use base 'y2_installbase';
+use strict;
+use warnings;
+use testapi qw(save_screenshot get_var);
+
+sub run {
+    my $timeout = 60 * get_var('TIMEOUT_SCALE', 1);
+    my $regcode = get_var('SCC_REGCODE_HA');
+    $testapi::distri->get_module_ha_regcode()->add_ha_registration_code($regcode, $timeout);
```

Comment:
the method that we expose from the test is `wait_ha_regcode_page`,  where is expected a parameter like timeout, but for `add_ha_registration_code` is not expected, looks more like a method that we hacked instead of something explicit.